api/resources_portal/views/material_request.py

```diff
@@ -0,0 +1,176 @@
+        else:
+            permission_classes = [
+                IsAuthenticated,
+            ]
+
+        return [permission() for permission in permission_classes]
+
+    # Determines what the next requirement(s) are for the material request.
+    # Returns a list of the next requirements, which is null if there are none left
+    def get_material_requirements(self):
+        request = self.get_object()
+        material = request.material
+
+        requirements_list = []
+
+        if request.status == "PENDING":
+            requirements_list.append("NEEDS_TO_BE_APPROVED")
+
+        if material.needs_mta and not request.requester_signed_mta_attachment:
+            requirements_list.append("NEEDS_MTA_TO_BE_SIGNED")
+
+        if material.needs_mta and not request.executed_mta_attachment:
+            requirements_list.append("NEEDS_MTA_TO_BE_SIGNED")
+
+        if material.needs_irb and not request.irb_attachment:
+            requirements_list.append("NEEDS_IRB")
+
+        return requirements_list
+
+    def create(self, request, *args, **kwargs):
+        serializer = MaterialRequestSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        material = serializer.validated_data["material"]
+        serializer.validated_data["assigned_to"] = material.contact_user
+
+        material_request = MaterialRequest(**serializer.validated_data)
+        material_request.save()
+
+        notification = Notification(
+            notification_type="TRANSFER_REQUESTED",
+            notified_user=material_request.assigned_to,
+            associated_user=request.user,
+            associated_material=material,
+        )
+        notification.save()
+
+        return Response(data=model_to_dict(material_request), status=201)
+
+    def update(self, request, *args, **kwargs):
+        material_request = self.get_object()
+
+        if material_request.status == "CANCELLED" or material_request.status == "REJECTED":
+            return Response(status=403)
+
+        serializer = MaterialRequestSerializer(material_request, data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        if request.user == material_request.requester:
+            if "irb_attachment" in request.data:
+                material_request.irb_attachment = serializer.validated_data["irb_attachment"]
+            if "requester_signed_mta_attachment" in request.data:
+                material_request.requester_signed_mta_attachment = serializer.validated_data[
+                    "requester_signed_mta_attachment"
+                ]
+            if "status" in request.data and request.data["status"] != material_request.status:
+                if serializer.validated_data["status"] != "CANCELLED":
+                    return Response(status=403)
+                else:
+                    material_request.status = serializer.validated_data["status"]
+        else:
+            if "status" in request.data:
```

Comment:
Can we also check that `request.data["status"] != "CANCELLED"`? I'd like for that state to only be set by the requester, otherwise the request should be `REJECTED`.

Reply:
sure! done.